src/main/java/org/icgc/argo/program_service/grpc/interceptor/EgoAuthInterceptor.java

```diff
@@ -50,7 +51,7 @@ public EgoAuthInterceptor(@NonNull EgoSecurity egoSecurity) {
     Metadata metadata,
     ServerCallHandler<ReqT, RespT> next) {
     String token = metadata.get(JWT_METADATA_KEY);
-    val egoToken = egoSecurity.verifyToken(token);
+    Optional<EgoToken> egoToken = egoSecurity.verifyToken(token);
```

Comment:
any reason why you made this not final?

Reply:
Comparing to val, it gives better readability.

Reply:
Do you mind changing it to `val`, because `val` is equivalent to `final Optional<EgoToken>`, the semantics are different the guarantee to the compiler is stronger.

Reply:
sure